### DISCUSSION TITLE
Fall Back to Basic/Built-In Autocomplete When No Language Server?

### DISCUSSION THREAD

**Discussion by treatmesubj** [hrsh7th/nvim-cmp#1751] [Q&A]
Hi, I use lsp-zero and nvim-cmp as recommended in their docs. I enjoy `Ctrl+N` autocomplete when I am working on a filetype for which I have a language server installed. However, when I work on a file like markdown, for which I do not have a language server installed, I wish that I had the old-school basic/built-in autocomplete functionality of `Ctrl+N`, however it doesn't seem to work or do anything anymore.

Is it possible to configure this?

Here is the top of my `~/.vimrc`...

```vim
" ~/.vimrc
call plug#begin('~/.vim/plugged')
Plug 'patstockwell/vim-monokai-tasty'
Plug 'vim-python/python-syntax'
Plug 'jlanzarotta/bufexplorer'
Plug 'mogelbrod/vim-jsonpath'
Plug 'itchyny/lightline.vim'
Plug 'treatmesubj/rock-lightline'
if has('nvim')
  Plug 'wookayin/semshi'
  " LSP Support
  Plug 'neovim/nvim-lspconfig'
  " Autocompletion
  Plug 'hrsh7th/nvim-cmp'
  Plug 'hrsh7th/cmp-nvim-lsp'
  Plug 'L3MON4D3/LuaSnip'
  " LSP-Zero
  Plug 'VonHeikemen/lsp-zero.nvim', {'branch': 'compat-07'}  " Debian has old nvim
  Plug 'nvim-treesitter/nvim-treesitter', {'do': ':TSUpdate'}
  " :TSInstall yaml
  Plug 'cuducos/yaml.nvim'
endif
call plug#end()
```

and here's top of my `~/.config/nvim/init.lua`...

```lua
-- ~/.config/nvim/init.lua
vim.api.nvim_command('set runtimepath^=~/.vim runtimepath+=~/.vim/after')
vim.api.nvim_command('let &packpath = &runtimepath')
vim.api.nvim_command('source ~/.vimrc')

local lsp_zero = require('lsp-zero')
lsp_zero.on_attach(function(client, bufnr)
  lsp_zero.default_keymaps({buffer = bufnr})
end)
-- show most severe errors on top
vim.lsp.handlers["textDocument/publishDiagnostics"] = vim.lsp.with(
    vim.lsp.diagnostic.on_publish_diagnostics, {
        severity_sort = true
    }
)
-- https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md
-- https://github.com/python-lsp/python-lsp-server
require('lspconfig').pylsp.setup{
  cmd = {(os.getenv("HOME")..'/.venv_pynvim/bin/pylsp')};
  settings = {
    pylsp = {
      plugins = {
        jedi = {
          environment = (os.getenv("HOME")..'/.venv_pynvim/bin/python')
        }
      }
    }
  },
  -- https://github.com/SmiteshP/nvim-navic#%EF%B8%8F-setup
  -- on_attach = function(client, bufnr)
  --   navic.attach(client, bufnr)
  -- end
}
```

**Comment by fitrh** [comment]
I don't know about lsp-zero, but all you need is [cmp-buffer](https://github.com/hrsh7th/cmp-buffer) and [cmp-path](https://github.com/hrsh7th/cmp-path) sources

**Reply by treatmesubj** [reply]
thanks, will see if I can implement [hrsh7th/cmp-buffer](https://github.com/hrsh7th/cmp-buffer). Adding it to `init.lua` doesn't seem to immediately work in my current configuration, but probably something else goofy going on.

**Comment by treatmesubj** [comment]
Whoops, actually the exact cmp-buffer source I needed was described right in their [docs](https://github.com/VonHeikemen/lsp-zero.nvim/blob/v3.x/doc/md/autocomplete.md#adding-a-source)!
